docker/documentation-site-builder/Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM ruby:3.1-slim-bullseye as jekyll
+
+SHELL ["/bin/bash", "-eo", "pipefail", "-c"]
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    git \
+    && rm -rf /var/lib/apt/lists/*
+
+# used in the jekyll-server image, which is FROM this image
+COPY docker-entrypoint.sh /usr/local/bin/
+
+RUN gem update --system && gem install jekyll && gem cleanup
```

Comment:
We should use the `Gemfile` and the `Gemfile.lock` here. Otherwise, this will always install the latest version of Jekyll when building the image.

Reply:
Done